modules/color-picker/react/spec/ColorInput.spec.tsx

```diff
@@ -1,115 +1,150 @@
-    );
-    const event = {
-      target: {value: '#e6e'},
-    };
-    const input = component.find('input');
-    input.simulate('change', event);
-    expect(onValidColorChangeMock).toBeCalledWith('#ee66ee');
-  });
-  test('should not call onValidColorChange prop onChange if hex code is not valid', () => {
-    const onValidColorChange = jest.fn();
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={onValidColorChange} value={''} />
-    );
-    const event = {
-      target: {value: 'e6ee'},
-    };
-    const input = component.find('input');
-    input.simulate('change', event);
-    expect(onValidColorChange).toHaveBeenCalledTimes(0);
-  });
-  test('should remove hash', () => {
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#e6e'} />
-    );
-    expect(component.find('input').props().value).toBe('e6e');
-  });
-  test('should allow only 6 non-# characters as the value', () => {
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#123456789'} />
-    );
-    expect(component.find('input').props().value).toBe('123456');
-  });
-  test('should show the checkmark after the valid hex value has been formatted', () => {
-    const component = mount(
-      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#123456789'} />
-    );
-    expect(
-      component
-        .find('div[className^="css"]')
-        .at(1) // This div the background of the color preview/checkmark bit
-        .prop('style')
-    ).toHaveProperty('backgroundColor', '#123456');
-  });
-  test('should set inputRef as reference to native input', () => {
-    const ref = React.createRef<HTMLInputElement>();
-    mount(<ColorInput inputRef={ref} value={''} />);
-    expect(ref.current && ref.current.tagName && ref.current.tagName.toUpperCase()).toBe('INPUT');
-  });
+  describe('when rendered', () => {
+    describe('with a placeholder', () => {
+      test('should render a ColorInput with a placeholder', () => {
+        const {getByTestId} = render(
+          <ColorInput onChange={cb} placeholder={placeholder} data-testid={id} />
+        );
+        expect(getByTestId(id)).toHaveAttribute('placeholder', placeholder);
+      });
+    });
+
+    describe('with an id', () => {
+      test('should render a ColorInput with an id', () => {
+        const {getByTestId} = render(<ColorInput id="my-id" data-testid={id} />);
+        expect(getByTestId(id)).toHaveAttribute('id', 'my-id');
+      });
+    });
+
+    describe('with a value', () => {
+      test('should render a ColorInput with a value', () => {
+        const {getByTestId} = render(<ColorInput value={value} data-testid={id} />);
+        expect(getByTestId(id).value).toBe(value);
+      });
+    });
+
+    describe('with a value and check', () => {
```

Comment:
Not a comment on this test but I guess while you're in here with the stories, it may be good to add a story, example, or knob for the checked state, I realized that isn't demonstrated at all in our Storybook. It could also be a separate PR as well.

Reply:
Yup, added a `Checked` story 😎